### PR TITLE
[Feature] Add API for speech synthesis

### PR DESCRIPTION
**Pull request by dcvz**

### CODE REVIEW

File: src/speechsynthesizer/DarwinSpeechSynthesizer.hpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+public:
+    DarwinSpeechSynthesizer();
+
+    void Speak(const char* text);
+
+protected:
+    bool DoInit(void);
+
+private:
+    void *mSynthesizer;
```
Comment (Kenix3) [inline src/speechsynthesizer/DarwinSpeechSynthesizer.hpp]:
```suggestion
    void* mSynthesizer;
```

**Review by Kenix3 — CHANGES_REQUESTED** [review]
Only other comment is that we use `h` instead of `hpp`. We definitely don't want the bridge to be `hpp`.

File: src/speechsynthesizer/DarwinSpeechSynthesizer.hpp (hunk 1 above)
Comment (dcvz) [reply src/speechsynthesizer/DarwinSpeechSynthesizer.hpp]: This was changed by clang tidy as far as I remember.

Comment (Kenix3) [reply src/speechsynthesizer/DarwinSpeechSynthesizer.hpp]: It really should have failed the clang-format test actually.

Comment (dcvz) [reply src/speechsynthesizer/DarwinSpeechSynthesizer.hpp]: might be because of the hpp and h. it might only check .h files? renaming them now.

Comment (dcvz) [reply src/speechsynthesizer/DarwinSpeechSynthesizer.hpp]: yup, that was it! it now catches things in the h files

**Review by Kenix3 — COMMENTED** [review]
I'm a little nervous that there is no Linux synthesizer, but I'm not sure if it needs to hold up this PR either.

**Pull request merged**